react-redux/react-redux.d.ts

```diff
@@ -7,39 +7,68 @@
 /// <reference path="../redux/redux.d.ts" />
 
 declare module "react-redux" {
-  import { ComponentClass, Component, StatelessComponent } from 'react';
+  import { ComponentClass, Component, StatelessComponent, Props, ReactNode } from 'react';
   import { Store, Dispatch, ActionCreator } from 'redux';
 
-  export interface ComponentConstructDecorator<P> {
-    <TComponentConstruct extends (ComponentClass<P>|StatelessComponent<P>)>(component: TComponentConstruct): TComponentConstruct
+  /** Generic decorator, that receives T = original props, U = own props */
+  interface ComponentDecorator<T extends Props<any>, U extends Props<any>> {
```

Comment:
I think it's better for necessary information to be documented in the parameter itself than in comment if possible.
e.g. 

```ts
interface ComponentDecorator<
  TConnectedProps extends Props<any>,
  TOwnProps extends Props<any>
> {
  (component: ComponentClass<TConnectedProps>): ComponentClass<TOwnProps>;
}
```